File: KanbanBoardCore/Project.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace KanbanBoardCore
{
    public class Project : CoreObject
    {
        private DateTime? _completionDate = null;
        public int ProjectID { get; set; }

        public string ProjectName { get; set; }

        public string Description { get; set; }

        public DateTime StartDate { get; set; }

        public DateTime DueDate { get; set; }

        public DateTime? CompletionDate
        {
            get
            {
                if (_completionDate != null && _completionDate.Value == DateTime.MinValue)
                    return null;
                return _completionDate;
            }

            set
            {
                _completionDate = value;
            }
                
        }

    }
}

File: KanbanUI/ClientApp/src/app/modals/Project.ts
export class ProjectStage{
    StageID : number;
    StageName: string;
    ProjectID : number;
    Order : number;
}

export class Project{
    ProjectID : number;
    ProjectName : string;
    Description: string;
    Quadrant:boolean;
    StartDate : Date;
    DueDate : Date;
    Completion: Date;
    Stage : ProjectStage[];  

    public constructor(init?:Partial<Project>){
        Object.assign(this,init);
    }
}

export class Priority{
    PriorityID : number;
    PriorityName : string;
}

export class ProjectTask{
    TaskID : number = 0;
    Summary : string = "";
    Description: string = "";
    PriorityID:number = 0;
    StageID : number = 0;
    DueDate : Date = null;
    CompletionDate : Date = null;

    public constructor(init?:Partial<ProjectTask>){
        Object.assign(this, init);
    }
}
